drivers/flash/Kconfig.simulator

```
@@ -26,6 +26,14 @@ config FLASH_SIMULATOR_DOUBLE_WRITES
 	 If selected, writing to a non-erased program unit will succeed, otherwise, it will return an error.
 	 Keep in mind that write operations can only pull bits to zero, regardless.
 
+config FLASH_SIMULATOR_NOALLOC
```

depends on !NATIVE_POSIX

---

!ARCH_POSIX you mean ?

---

As I mentioned in the review, this setting changes the simulator into a stub, so please change the name.

That way a user immediately understands that this will change behavior.

Especially, the name `NOALLOC` together with `will cause driver to allocate only write_block_size`.
actually means it will allocate memory, just less memory.

```suggestion
config FLASH_SIMULATOR_STUB
```